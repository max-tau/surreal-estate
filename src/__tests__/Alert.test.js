import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays an error message", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);

    expect(getByText(/Error/).textContent).toBe("Error!");
    expect(asFragment()).toMatchSnapshot();
  });
  it("displays a success message", () => {
    const { getByText, asFragment } = render(<Alert message="Success!!!" />);

    expect(getByText(/Success/).textContent).toBe("Success!!!");
    expect(asFragment()).toMatchSnapshot();
  });
  it("does not render without a message", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
