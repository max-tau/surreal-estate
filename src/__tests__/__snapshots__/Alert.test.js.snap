// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render without a message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  />
</DocumentFragment>
`;
